### PR TITLE
Refactor parse method to fix rubocop issues

### LINKED ISSUE #252: stats/perf-stat: resolve rubocop Metrics issues of parse method
### Reproduce
```
$ rubocop --only Metrics/AbcSize,Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity stats/perf-stat
stats/perf-stat:141:1: C: Metrics/AbcSize: Assignment Branch Condition size for parse is too high. [<36, 36, 22> 55.46/30]
def parse ...
^^^^^^^^^
stats/perf-stat:141:1: C: Metrics/CyclomaticComplexity: Cyclomatic complexity for parse is too high. [17/7]
def parse ...
^^^^^^^^^
stats/perf-stat:141:1: C: Metrics/PerceivedComplexity: Perceived complexity for parse is too high. [18/8]
def parse ...
^^^^^^^^^
```
### Reference
https://github.com/intel/lkp-tests/wiki/Coding-Standard#rubocop

### Acceptance Test
```
$ rubocop --only Metrics/AbcSize,Metrics/CyclomaticComplexity,Metrics/MethodLength,Metrics/PerceivedComplexity stats/perf-stat # has no issue of parse related code
$ rake spec spec=stats # pass
```

### PR DESCRIPTION
**Pull request by fongjiantan**
Fixes #252

### CODE REVIEW

File: stats/perf-stat [hunk 1]
```diff
@@ -219,6 +193,64 @@ def parse
...
+  if fields[0][0] == 'S'
+    socket = fields[0]
+    fields.delete_at 1
+    fields.delete_at 0
+  end
+
+  socket
+end
+
+def parse_unit(fields)
```
Comment (rli9) [inline stats/perf-stat:209]: thanks, i think the code can be one line as below
```
fields[1] if fields[1] == 'Bytes'
```

File: stats/perf-stat [hunk 2]
```diff
@@ -219,6 +193,64 @@ def parse
   # skip the last record, because the interval hasn't run out
 end
 
+def parse_per_socket_mode(fields)
+  # S0,12,1202404,,mem_load_retired.l1_miss,8004582139,66.65,,
+  socket = nil
```
Comment (rli9) [inline stats/perf-stat:198]: probably here can do early return like
```
return nil unless fields[0][0] == 'S'
```

File: stats/perf-stat [hunk 3]
```diff
@@ -219,6 +193,64 @@ def parse
...
+  unit = nil
+
+  unit = fields[1] if fields[1] == 'Bytes'
+
+  unit
+end
+
+def parse_metrics(metric_val, fields, socket)
+  # 1.000679357,S0,12,170516599,,l1d_pend_miss.pending,8004582139,66.65,80.5,Load_Miss_Real_Latency
+  metric_key = nil
```
Comment (rli9) [inline stats/perf-stat:220]: similarly, here can consider early return

File: stats/perf-stat [hunk 4]
```diff
@@ -219,6 +193,64 @@ def parse
...
+    metric_key = "#{socket}.#{metric_key}" if socket
+    metric_val = fields[4]
+    metric_val = metric_val.to_f
+  end
+
+  [metric_key, metric_val]
+end
+
+def update_keys(unit, key, socket)
+  socket_key = nil
```
Comment (rli9) [inline stats/perf-stat:233]: thanks, this line is not needed. The later code ```socket_key = "#{socket}.#{key}" if socket``` will set socket_key as nil if socket is nil.

**Comment by fongjiantan** [comment]
@rli9 I have updated the code based on your review. Thanks

**Comment by rli9** [comment]
thanks for the patch

**Pull request merged**